File: 1140.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner t = new Scanner(System.in);

		while (t.hasNext()) {
			
			String N = t.nextLine();
			
			if (N.equalsIgnoreCase("*"))
				break;
			int cont = 0;

			String[] S = N.split(" ");

			int i = 1;
			String aux = String.valueOf(S[0].charAt(0));
			while (i < S.length) {
				if (aux.equalsIgnoreCase(String.valueOf(S[i].charAt(0)))) {
					cont++;
				}
				aux = String.valueOf(S[i].charAt(0));
				i++;
			}
			if (cont == S.length - 1)
				System.out.println("Y");
			else
				System.out.println("N");
		}
	}
}

File: 1140.py
# -*- coding: utf-8 -*-

leitor = '1'
while leitor != '*':
    leitor = raw_input()

    aux = leitor[0].upper()

    cont = 0
    if leitor != '*':
        leitor = leitor .split()
        for i in xrange(len(leitor)):
            if leitor[i][0].upper() == aux:
                cont +=1
        if cont == len(leitor):
            print 'Y'
        else:
            print 'N'
